### PR TITLE
Merge UserStorageIterator with RecursiveNodeIterator, don't use filter iterator

### PR DESCRIPTION
**Pull request by IljaN**
## Related Issue
See #29 
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by IljaN** [comment]
- Add unit-tests

File: lib/Exporter/FilesExporter.php [hunk 1]
```diff
@@ -51,7 +51,7 @@ public function export(string $userId, string $exportPath) {
 		$userFolder = $this->rootFolder->getUserFolder($userId)->getParent();
 
 		/** @var \OCP\Files\Node $node */
-		foreach (new UserStorageIterator($userFolder) as $node) {
+		foreach (RecursiveNodeIterator::create($userFolder) as $node) {
```
Comment (jvillafanez) [inline lib/Exporter/FilesExporter.php]: I don't think this provides any advantage. We need a factory to be injected to be able to mock the iterator.

File: lib/Exporter/RecursiveNodeIterator.php [hunk 2]
```diff
@@ -22,25 +22,98 @@
...
 	private $currentIndex;
+	/** @var string[]  */
+	private $ignorePaths = [
+		'cache',
+		'thumbnails',
+		'uploads'
+	];
 
+	/**
+	 * @param \OCP\Files\Node[] $nodes
```
Comment (jvillafanez) [inline lib/Exporter/RecursiveNodeIterator.php]: Move the `Node` up to the "use" statement?

File: lib/Exporter/RecursiveNodeIterator.php [hunk 3]
```diff
@@ -22,25 +22,98 @@
...
+		foreach ($this->ignorePaths as $ignorePath) {
+			if (\substr($relativePath, 0, \strlen($ignorePath)) === $ignorePath) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	private function isUserStorage(IStorage $storage) : bool {
```
Comment (jvillafanez) [inline lib/Exporter/RecursiveNodeIterator.php]: does this work properly with an objectstore as primary storage?

**Comment by IljaN** [comment]
Obsoleted via #33

**Pull request closed without merging**